src/map/ai/controllers/mob_controller.cpp

```diff
@@ -720,6 +720,26 @@ void CMobController::FaceTarget(uint16 targid)
     }
 }
 
+bool CMobController::IsStuck()
+{
+    return m_Stuck;
+}
+
+void CMobController::UpdateLastKnownPosition()
+{
+    // Mob is considered "Stuck" if:
+    // 1. Last Pos && Current Pos are <= 1.5
+    // 2. Distance to Target > Melee Range
+    // 3. Mob is not bound or asleep
+    m_Stuck =
+        PMob->CanMove() &&
+        !PMob->isAsleep() &&
```

Comment:
I think this needs to be:
```suggestion
        !PMob->IsAsleep() &&
```
What do you think? isAsleep looks if your state is inactive, IsAsleep looks if you have Lullaby || Sleep || ...

Reply:
I see the confusion, the `IsAsleep` function you're referencing is part of the status effect container, not the mob entity.  Arguably it belongs on mob entity.

Reply:
Sorry, I guess I'm asking:
```
!PMob->StatusEffectContainer->IsAsleep()
```